### PR TITLE
Update DGP plugin mode flags

### PR DESCRIPTION
**Pull request by adam-enko**
Use `org.jetbrains.dokka.experimental.gradle.pluginMode=[V1Enabled|V2EnabledWithHelpers|V2Enabled]` to control the behaviour.

Use `org.jetbrains.dokka.experimental.gradle.pluginMode.noWarn=[true|false]` to enable/disable the warning logs.

KT-71236

### CODE REVIEW

File: dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt [hunk 1]
```diff
@@ -25,62 +27,102 @@ import java.util.*
...
+    private val pluginMode: PluginMode by lazy {
+        val parameterValue = parameters.pluginMode.getOrElse(PluginMode.Default.name)
+
+        val value = PluginMode.findOrNull(parameterValue)
+            ?: run {
+                logger.warn("Invalid value for $PLUGIN_MODE_FLAG. Got '$parameterValue' but expected one of: ${PluginMode.values.map { it.name }}")
+                PluginMode.Default
+            }
+
+        logger.info { "Detected PluginMode: $parameterValue/$value in ${parameters.projectDisplayName.orNull} ${parameters.projectDirectory.orNull}" }
```
Comment (whyoleg) [inline dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt:102]: minor: if I remember correctly, in `info` mode there will be no mention of `logger.name` and so no mention of Dokka, am I right?
also, minor, but AFAIR Gradle logs project name if there are logs, so this additional information about project is redundant

```suggestion
        logger.info { "Detected Dokka Gradle Plugin mode: $parameterValue/$value in ${parameters.projectDisplayName.orNull} ${parameters.projectDirectory.orNull}" }
```

File: dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt [hunk 2]
```diff
@@ -25,62 +27,102 @@ import java.util.*
...
 
     /**
      * Whether DGP should use V2 [org.jetbrains.dokka.gradle.DokkaBasePlugin].
      *
-     * Otherwise, fallback to V1 [org.jetbrains.dokka.gradle.DokkaClassicPlugin].
+     * @see pluginMode
      */
     internal val v2PluginEnabled: Boolean by lazy {
-        val v2PluginEnabled = parameters.v2PluginEnabled.getOrElse(false)
+        when (pluginMode) {
```
Comment (whyoleg) [inline dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt:78]: minor: would be nice to have `v2PluginMigrationHelpersEnabled` flag located just below `v2PluginEnabled` for easier understanding during reading of the code

**Review by whyoleg — APPROVED** [review]
Nice!
just some minor comments, can be merged even without them

File: dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt (hunk 2 above)
Comment (adam-enko) [reply dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt:78]: I agree, done.

File: dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt (hunk 1 above)
Comment (adam-enko) [reply dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt:102]: Good idea, done.

> AFAIR Gradle logs project name if there are logs

I think not. This is what I see with `./gradlew dokkaGenerate --dry-run --debug`

```
2024-09-23T13:16:08.596+0200 [WARN] [org.jetbrains.dokka.gradle.internal.PluginFeaturesService] Dokka Gradle Plugin PluginMode: V2Enabled/V2Enabled in root project 'KmpCommonSourceSharedWithDependentsTest' /Users/dev/projects/jetbrains/dokka/dokka-runners/dokka-gradle-plugin/build/test-temp-dir/functional-tests/KmpCommonSourceSharedWithDependentsTest
```

Comment (whyoleg) [reply dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt]: Ah, okay. Probably it happens only during configuration. I was meant this:
image

**Comment by adam-enko** [comment]
I'm waiting to find someone from KBT to review... in the interim I'll assign @Tapchicoma.

**Pull request merged**